### PR TITLE
Run stubtest on Windows for waitress

### PR DESCRIPTION
**Pull request by Avasam**
Yet another stub that benefits from #8923 . Fixing stubtest failure on Windows and offering better coverage.

I did not add `darwin` to `platforms` because it would be redundant with `linux`. But I still added a `stubtest_allowlist_darwin.txt` for those who would like to run it locally on macos. (This is the edge case mentioned in #9173 )

### CODE REVIEW

File: stubs/waitress/waitress/wasyncore.pyi [hunk 1]
```diff
@@ -77,20 +78,21 @@ class dispatcher_with_send(dispatcher):
...
-    def __init__(self, fd: BytesIO) -> None: ...
-    def __del__(self) -> None: ...
-    def recv(self, *args: Any) -> bytes: ...
-    def send(self, *args: Any) -> bytes: ...
-    def getsockopt(self, level: int, optname: int, buflen: bool | None = ...) -> int: ...
-    read: Callable[..., bytes] = ...
-    write: Callable[..., bytes] = ...
-    def close(self) -> None: ...
-    def fileno(self) -> BytesIO: ...
+if sys.platform == "linux" or sys.platform == "darwin":
```
Comment (JelleZijlstra) [inline stubs/waitress/waitress/wasyncore.pyi:81]:
```suggestion
if sys.platform != "win32":
```
Consistent with the other check you added.

Comment (Avasam) [reply stubs/waitress/waitress/wasyncore.pyi:81]: When the original code does `if os.name == "posix"`, the stub did `if sys.platform == "linux" or sys.platform == "darwin"` (there's another platform check like this in `waitress.trigger`).

I added `if sys.platform != "win32"` to be consistent with other `if hasattr(socket, "AF_UNIX")` checks.

But I can change them all to be either if you'd like.

Comment (JelleZijlstra) [reply stubs/waitress/waitress/wasyncore.pyi:81]: I think the runtime checks are pretty much equivalent, and in stubs we have tended to assume all non-Windows OSes are Unix.

Comment (Avasam) [reply stubs/waitress/waitress/wasyncore.pyi:81]: Sounds good, I'll make all 3 `!= "win32`

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**